app/gui2/src/util/colors.ts

```diff
@@ -0,0 +1,31 @@
+import { formatCss, inGamut, toGamut } from 'culori'
+import { v3 as hashString } from 'murmurhash'
+
+const inOklch = inGamut('rgb')
+const toOklch = toGamut('rgb')
+
+/* Generate a sRGB color value from the provided string. */
+export function colorFromString(s: string) {
+  const hash: number = hashString(s)
+  // Range values below can be adjusted if necessary, they were chosen arbitrarily.
+  const hue = mapInt32(hash, 0, 360)
+  const chroma = mapInt32(hash, 0.05, 0.14)
+  const lightness = mapInt32(hash, 0.52, 0.57)
+  let color = {
+    mode: 'oklch',
+    l: lightness,
+    c: chroma,
+    h: hue,
+  }
+  if (!inOklch(color)) {
+    color = toOklch(color)
+  }
+  return formatCss(color)
```

Comment:
We should really be using the actual `oklch` CSS color here. Otherwise we are throwing away all colors supported by displays of wider gamut than sRGB (such as all modern macs, which support P3). The browser is supposed to do a good job of mapping impossible colors to a decent fallback in the display gamut.

https://drafts.csswg.org/css-color/#css-gamut-mapping

```suggestion
  return `oklch(${lightness}%,${chroma},${hue})`
```

Reply:
Firstly, I was acting on the assumption the browser *will not* apply a correct gamut mapping. I saw this phrase in `oklch` [intro](https://evilmartians.com/chronicles/oklch-in-css-why-quit-rgb-hsl):

> The CSS Colors 4 spec [requires](https://www.w3.org/TR/css-color-4/#css-gamut-mapping) browsers to use the OKLCH method for gamut mapping. But still, right now, Chrome and Safari use the fast, but inaccurate, clipping method. That’s why we currently recommend manual gamut mapping and adding both sRGB and P3 colors to CSS:

I found no issue or information on the algorithm used in recent Chrome, though. And I guess this is not a big deal anyway, and we can safely rely on the browsers' default behavior.

Secondly, we lose the support of older browsers here by using `oklch` in CSS. We can avoid that by translating the color to sRGB (and doing a gamut mapping by clamping), but this way, we will lose a wider range of P3 colorspace.

So the question is: do we want to support older browsers and sacrifice color range? (`oklch` is still useful for accurate color generation, though)

Reply:
There is also another option. You can keep the `oklch` as the preffered style, but also add a fallback style. In CSS, this can be done like
```css
.selector {
    color: rgb(1,2,3);
    color: oklch(1,2,3);
}
```
Browsers will apply the last valid property, and `oklch` is considered invalid in browsers that don't support it. I believe that our current CSS preprocessing setup will already do that automatically for statically defined styles, but it's worth checking.

For dynamic properties it's a bit more tricky, as you have to do the conversion yourself. I wonder if we could somehow make Vue handle that automatically as well, since it does it for vendor prefixes already anyway.

Reply:
Note that Vue accepts arrays as style property definitions. that way you can provide multiple dynamic variants of a style, and the last valid one wins. So, to implement color fallback, we could do something like:

```ts
const rgbCss = fallbackRgb(oklch)
return [rgbCss, oklchCss]
```

Reply:
The proposed solution didn't work (easily), so after discussion we decided to just check `oklch` support when generating colors.